org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/services/TextDocumentService.java

```diff
@@ -150,7 +151,7 @@
 	 * Registration Options: TextDocumentRegistrationOptions
 	 */
 	@JsonRequest
-	CompletableFuture<List<? extends SymbolInformation>> documentSymbol(DocumentSymbolParams params);
+	CompletableFuture<Either<List<? extends SymbolInformation>, List<? extends DocumentSymbol>>> documentSymbol(DocumentSymbolParams params);
```

Comment:
I doubt that the current implementation of Either is able to parse this return type. Could you add a test to JsonParseTest that checks parsing a `documentSymbol` ResponseMessage? If it fails, please mark it with `@Ignore` and create an issue for that problem. Then we can merge this PR anyway and take care of the parsing later.